### PR TITLE
Add tooltips to git views

### LINKED ISSUE #81: Include a tooltip for every icon without a label
#### Description
Opened a file, saw this in the bottom-right corner:

huh
Had no idea what it was trying to tell me, and no tooltip was available, either. I resorted to kicking open the developer tools, inspecting the icon in the console, and checking the class attribute to ascertain its purpose.

I'd add a tooltip for that icon, as well as anything else that indicates something that mightn't be immediately apparent to the user. Sure, it might be obvious after clicking it, but it's generally not good practice in UX to encourage users to press something in order to understand its purpose.

erm

I'd also make the deprecation cop use the same visual cue (pointer cursor, underlined text, etc) as the language and encoding menus. I understand that's a menial issue, though, and one that's better off being filed as a separate issue in the appropriate repo.

#### Specs
 - **Platform:** Mac OS/X 10.9.5 (Screw Yosemite)
 - **Processor:** 2.3 GHz Intel Core i7
 - **Memory:** 16 GB 1600 MHz DDR3
 - **Atom version:** 0.206.0

### PR DESCRIPTION
**Pull request by Tyriar**
Fixes #81 

status_bar_tooltips

### CODE REVIEW

**Comment by Tyriar** [comment]
There didn't seem to be any specs in the package for the existing tooltips that were triggered via hover, any thoughts on the best approach to test this?

**Comment by steelbrain** [comment]
:heart:

File: lib/git-view.coffee [hunk 1]
```diff
@@ -117,12 +123,16 @@ class GitView extends HTMLElement
       if ahead > 0
         @commitsAhead.textContent = ahead
         @commitsAhead.style.display = ''
+        @commitsAheadTooltipDisposable?.dispose()
+        @commitsAheadTooltipDisposable = atom.tooltips.add @commitsAhead, title: "#{ahead} commits ahead of upstream"
```
Comment (kevinsawicki) [inline lib/git-view.coffee]: This should probably use something like `_.pluralize` so that it doesn't end up as `1 commits`

Comment (Tyriar) [inline lib/git-view.coffee]: Used `_.pluralize` for line(s) and commit(s)

File: lib/git-view.coffee [hunk 2]
```diff
@@ -117,12 +124,16 @@ class GitView extends HTMLElement
       if ahead > 0
         @commitsAhead.textContent = ahead
         @commitsAhead.style.display = ''
+        @commitsAheadTooltipDisposable?.dispose()
+        @commitsAheadTooltipDisposable = atom.tooltips.add @commitsAhead, title: "#{_.pluralize(ahead, 'commit')} commits ahead of upstream"
```
Comment (50Wliu) [inline lib/git-view.coffee]: I don't think you need the commits after the pluralize here

File: lib/git-view.coffee [hunk 3]
```diff
@@ -137,15 +148,20 @@ class GitView extends HTMLElement
...
       stats = repo.getDiffStats(itemPath)
       if stats.added and stats.deleted
         @gitStatusIcon.textContent = "+#{stats.added}, -#{stats.deleted}"
+        tooltipText = "#{_.pluralize(stats.added, 'line')} added and #{_.pluralize(stats.deleted, 'line')} deleted in this file not yet committed"
       else if stats.added
         @gitStatusIcon.textContent = "+#{stats.added}"
+        tooltipText = "#{_.pluralize(stats.added, 'line')} added to this file not yet committed"
       else if stats.deleted
         @gitStatusIcon.textContent = "-#{stats.deleted}"
+        tooltipText = "#{_.pluralize(stats.deleted, 'line')} lines added from this file not yet committed"
```
Comment (50Wliu) [inline lib/git-view.coffee]: Ditto, lines not needed

**Comment by Tyriar** [comment]
Ping?

**Comment by joshaber** [comment]
This is something that's been bothering me too, thanks @Tyriar!

>There didn't seem to be any specs in the package for the existing tooltips that were triggered via hover, any thoughts on the best approach to test this?

Hmm, you could probably get some inspiration from the tooltip manager's specs: https://github.com/atom/atom/blob/afc32d5905f10c2c9124e142fa938aa402a9c04b/spec/tooltip-manager-spec.coffee#L16

**Comment by Tyriar** [comment]
@joshaber Added tests for the cases already covered, missing are `<num> commit(s) ahead/behind <branch>` and `<num> line(s) deleted`.

**Comment by joshaber** [comment]
Awesome, thanks!

**Comment by steelbrain** [comment]
:rocket: :rocket: :rocket: :rocket: :rocket: :rocket:

File: lib/git-view.coffee [hunk 4]
```diff
@@ -137,15 +148,20 @@ class GitView extends HTMLElement
...
       stats = repo.getDiffStats(itemPath)
       if stats.added and stats.deleted
         @gitStatusIcon.textContent = "+#{stats.added}, -#{stats.deleted}"
+        tooltipText = "#{_.pluralize(stats.added, 'line')} added and #{_.pluralize(stats.deleted, 'line')} deleted in this file not yet committed"
       else if stats.added
         @gitStatusIcon.textContent = "+#{stats.added}"
+        tooltipText = "#{_.pluralize(stats.added, 'line')} added to this file not yet committed"
       else if stats.deleted
         @gitStatusIcon.textContent = "-#{stats.deleted}"
+        tooltipText = "#{_.pluralize(stats.deleted, 'line')} added from this file not yet committed"
```
Comment (jerone) [inline lib/git-view.coffee]: Shouldn't "added" be "deleted"?

**Pull request merged**